packages/next/src/server/lib/cache-handlers/default.ts

```diff
@@ -105,22 +108,22 @@ const DefaultCacheHandler: CacheHandler = {
     }
   },
 
+  async refreshTags() {
+    // Nothing to do for an in-memory cache handler.
+  },
+
+  async getExpiration(...tags) {
+    return Math.max(...tags.map((tag) => tagsManifest.get(tag) ?? 0))
+  },
+
   async expireTags(...tags) {
+    const timestamp = Math.round(performance.timeOrigin + performance.now())
```

Comment:
I wonder if conceptually the cache handler needs to be run outside of all contexts. In theory we shouldn't need to work around sync dynamic things with stuff like timeOrigin.